### DISCUSSION TITLE
Controlling the summarizing limits on print for dataframes

### DISCUSSION THREAD

**Discussion by dnessett** [vaexio/vaex#1647] [Q&A]
I am trying to print out a dataframe's values without summarization. The dataframe has 15 rows and the interesting rows are buried in the "...". I tried using

```
with np.printoptions(threshold=np.inf):
#    print(osc_data_zc_clean_norm[41330:41345])
    print(period[41330:41345])
```

But this had no effect, i.e.

```
freq
#    sx                      x                        zc
0    0.00022656069681181859  -0.30971412713485447     True
1    0.00017453844030594344  -0.3097461698290791      True
2    0.0001730305488130195   -0.30977142701158555     True
3    0.00013156353275761178  -0.30977218095733206     True
4    0.0001112069976031389   -0.3097797204147967      True
...  ...                     ...                      ...
10   0.30837436554339154     -7.313273740680999e-05   True
11   0.30836607214018047     -0.00010856818749052204  True
12   0.308335914310322       -0.00011384580771575576  True
13   0.3082899236197878      -0.00013457931574345963  True
14   0.30827559865060505     -0.00018886340948872065  True
```

I couldn't find an equivalent print control documented for vaex dataframes. Is there any way to override the summarizing limits on print calls that works on dataframes?

**Comment by maartenbreddels** [comment]
Hi,

good question, I think https://vaex.io/docs/api.html#vaex.dataframe.DataFrame.head_and_tail_print or https://vaex.io/docs/api.html#vaex.dataframe.DataFrame.cat can help you
the treshold is about the number printing, vaex does not look at numpy for how many rows it should print.

cheers,

Maarten

**Reply by dnessett** [reply]
> Hi,
> 
> good question, I think https://vaex.io/docs/api.html#vaex.dataframe.DataFrame.head_and_tail_print or https://vaex.io/docs/api.html#vaex.dataframe.DataFrame.cat can help you the treshold is about the number printing, vaex does not look at numpy for how many rows it should print.
> 
> cheers,
> 
> Maarten

Thanks, Maarten. "cat" worked perfectly.

Dan
